apps/condo/domains/billing/schema/resolvers/propertyResolver.js

```diff
@@ -0,0 +1,357 @@
+                    this.regExps[search.slice(1)] = replace
+                } else {
+                    this.replaces[search] = replace
+                }
+            }
+        } catch {
+            // logging miss-configuration
+        }
+    }
+
+    apply (input) {
+        if (typeof input !== 'string') {
+            return input.trim()
+        }
+        for (const regFrom of Object.keys(this.regExps)) {
+            if (this.regExps.hasOwnProperty(regFrom)) {
+                const regTo = this.regExps[regFrom]
+                input = input.trim().replace(new RegExp(regFrom, 'g'), regTo)
+            }
+        }
+
+        for (const from of Object.keys(this.replaces)) {
+            if (this.replaces.hasOwnProperty(from)) {
+                input = input.trim().split(from).join(this.replaces[from])
+            }
+        }
+
+        return input
+            .replaceAll(/\s+/g, ' ')
+            .trim()
+    }
+
+}
+
+
+class PropertyFinder {
+
+    constructor () {
+        this.organizationProperties = []
+    }
+
+    init (organizationProperties) {
+        this.organizationProperties = organizationProperties.map(property => ({
+            ...property,
+            tokens: this.getTokensFromAddress(property.address),
+        }))
+    }
+
+    getTokensFromAddress (addressStr) {
+        const SYMBOLS_TO_REMOVE_REGEXP = /[!@#$%^&*)(+=_:"'`[\]]/g
+        const SPLITTERS_REGEXP = /[,;.\s-/]/
+        const IS_DIGITS_ONLY_REGEXP = /^\d+$/
+        return addressStr
+            .toLowerCase()
+            .replace(/[ёë]/g, 'е')
+            .replace(SYMBOLS_TO_REMOVE_REGEXP, '')
+            .replace(/\s+/g, ' ')
+            .split(SPLITTERS_REGEXP)
+            .filter(Boolean)
+            .filter((x) => x.length > 1 || IS_DIGITS_ONLY_REGEXP.test(x))
+    }
+
+    orderedIntersection (arr1, arr2) {
+        const result = []
+        const _arr2 = Array.from(arr2)
+        for (const str1 of arr1) {
+            if (_arr2.includes(str1)) {
+                result.push(str1)
+                _arr2.splice(0, _arr2.indexOf(str1) + 1)
+            }
+        }
+        return result
+    }
+
+    findPropertyByOrganizationAndAddress (address) {
+        const targetTokens = this.getTokensFromAddress(address)
+        let maxScore = 0
+        let theMostProbableProperties = []
+        for (const property of this.organizationProperties) {
+            const intersection = this.orderedIntersection(targetTokens, property.tokens)
+            const score = intersection.length
+            if (score >= maxScore) {
+                if (score === maxScore) {
+                    theMostProbableProperties.push(property)
+                } else {
+                    theMostProbableProperties = [property]
+                    maxScore = score
+                }
+            }
+        }
+        return theMostProbableProperties[0]
+    }
+
+}
+
+module.exports = {
+    AddressTransform,
```

Comment:
Omg! it's a new pattern! Other files return just one resolver! Why this file is not the same?